exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java

```diff
@@ -142,23 +140,37 @@ void submitValidTransaction(VertxTestContext context) {
                 })));
   }
 
+  @Test
+  void submitTransactionOfIncorrectMessageSize(VertxTestContext context) {
+    BinaryMessage message = getBinaryMessage();
+    byte errorByte = 1;
+
+    post(ApiController.SUBMIT_TRANSACTION_PATH)
+        .sendBuffer(
+            Buffer.buffer(message.getSignedMessage().array()).appendByte(errorByte),
```

Comment:
This method does not seem to test what it says it tests. As we use a mock of a service, `service.convertTransaction(message)` in the handler will return null (which is prohibited by the interface specification) — the additional byte won't change anything. I'll make this test fail _properly_ first to ensure that we get a correct error.

I suspect there either happens an NPE in the handler or a Mockito exception (because strict stubs are enabled).

Reply:
I see, it tests invariants internal to _any_ message, but not to a message of a _particular_ transaction (that first invariant is 'payload_size' having a correct value). Do you think anything could be clarified to avoid such confusion?